deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -1067,7 +1068,8 @@ def average_tensor(self, tensor):
 
             process_group = self.dp_process_group
             # count = 0
-            for i, param, param_id in self.params_in_ipg_bucket:
+            for i, param_idx_in_group, param_id in self.params_in_ipg_bucket:
+                param = self.bit16_groups[group_idx][param_idx_in_group]
```

Comment:
@nelyahu - this is failing on group_idx just fyi

Reply:
fixed